tools/proc_init.py

```diff
@@ -0,0 +1,302 @@
+    nodes_data = read_nodes_file( params.input_dir+'/'+params.nodes_file )
+    rcvProc_nodes_data = include_recv_proc_nodes( nodes_data, metis_partitions )
+    print( 'Rank: ', rank, ', Completed loading nodes data: ', rcvProc_nodes_data.shape )
+
+    rcvProc_edge_data = []
+    edges_data = read_edge_file( params.input_dir+'/'+params.edges_file )
+    rcvProc_edge_data = include_recv_proc_edges( edges_data, metis_partitions )
+    print( 'Rank: ', rank, ', Completed loading edges data: ', rcvProc_edge_data.shape )
+
+    node_features = []
+    node_features = read_node_features_file( params.input_dir+'/'+params.node_feats_file )
+    print( 'Rank: ', rank, ', Completed loading node features reading from file ', len(node_features))
+
+    edge_features = []
+    #edge_features = read_edge_features_file( params.input_dir+'/'+params.edge_feats_file )
+    #print( 'Rank: ', rank, ', Completed edge features reading from file ', len(edge_features) )
+
+    return rcvProc_nodes_data, node_features, rcvProc_edge_data, edge_features
+
+def run(rank, size, params):
+
+    metis_partitions = read_metis_partitions(params.input_dir+'/'+params.metis_partitions)
+    print( 'Rank: ', rank, ', Completed loading metis partitions: ', len(metis_partitions))
+
+    schema_map = read_json(params.input_dir+'/'+params.schema)
+    ntypes_map, ntypes = get_node_types(schema_map)
+
+    if rank == 0: 
+        node_data, node_features, edge_data, edge_features = readInputFiles( rank, params, metis_partitions )
+
+        # order node_data by node_type before extracting node features. 
+        # once this is ordered, node_features are automatically ordered and 
+        # can be assigned contiguous ids starting from 0 for each type. 
+        node_data = node_data[ node_data[:, 0].argsort() ]
+
+        print( 'Rank: ', rank, ', node_data: ', node_data.shape )
+        print( 'Rank: ', rank, ', node_features: ', len(node_features))
+        print( 'Rank: ', rank, ', edge_data: ', edge_data.shape )
+        #print( 'Rank: ', rank, ', edge_features : ',len( edge_features) )
+        print( 'Rank: ', rank, ', partitions : ', len(metis_partitions ))
+
+        # shuffle data
+        sendOutData( rank, node_data, node_features, edge_data, metis_partitions, ntypes_map)
+
+        # Filter data owned by rank-0
+        node_data = node_data[ :, [0,5,6] ][ node_data[ :, 7 ] == 0 ] #extract only ntype, orig_type_nid, line_id
+        edge_data = edge_data[ :, [0,1,2,3]][edge_data[:,4] == 0 ] #extract only orig-src-id, orig-dst-id, orig-type-id orig-type
+    else: 
+        node_data = recvNodeData (rank, 2, torch.int32)
+        node_features = recv_node_features_obj(rank, size)
+        edge_data = recvData( rank, 2, torch.int32 )
+
+    '''
+    sort data by type-id, and follow through with assigning globalIds
+    '''
+    type_ids = np.unique( node_data[ :, 0 ] )
+    type_ids.sort ()
+
+    lstNodeTypeCounts = []
+    typeCounts = np.bincount( node_data[ :, 0 ] )
```

Comment:
you don't need to call another numpy function to get the count. numpy unique can count the elements as well. you can check the api of numpy unique.